[Docs] Add warning about `stubtest` code execution (#13927)
python/mypy: docs/source/stubtest.rst

test Python's official collection of library stubs,
`typeshed <https://github.com/python/typeshed>`_.

.. warning::
    

>     stubtest attempts to import specified python modules.
>     It will execute all module level code. Please, keep this in mind.

Reviewer: ```suggestion
    stubtest attempts to import specified Python modules.
    It will execute all module level code.
```
"keep this in mind" doesn't really add anything

>     stubtest attempts to import specified python modules.
>     It will execute all module level code. Please, keep this in mind.

Reviewer 2: ```suggestion
    stubtest will execute Python code from the packages it checks.
```

Reviewer 2: stubtest can run a lot of code that is not module-level

Author: That's fair!
